src/main/java/hudson/plugins/helpers/AbstractBuildAction.java

```diff
@@ -51,6 +56,26 @@ public synchronized void setBuild(BUILD build) {
         }
     }
 
+    private BUILD runToBuild(Run<?, ?> run) throws IllegalStateException {
+        final BUILD b;
+        try {
+            b = (BUILD)run;

```

Comment:
This is misleading—the same as

```java
b = (AbstractBuild) run;
```

Reply:
Nope, because it's a generic: `AbstractBuildAction<BUILD extends AbstractBuild<?, ?>> `, which is being stored to a field: `private transient BUILD build = null;`. We could get rid of that by breaking binary compatibility, but I would rather avoid it

Reply:
The erasure is simply `AbstractBuild`.